imgui/src/extension_imgui.cpp

```diff
@@ -1236,6 +1236,96 @@ static int imgui_Checkbox(lua_State* L)
     return 2;
 }
 
+static int imgui_BeginMenuBar(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 1);
+    imgui_NewFrame();
+
+    bool result = ImGui::BeginMenuBar();
+    lua_pushboolean(L, result);
+
+    return 1;
+}
+
+static int imgui_EndMenuBar(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 0);
+    imgui_NewFrame();
+
+    ImGui::EndMenuBar();
+
+    return 0;
+}
+
+static int imgui_BeginMainMenuBar(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 1);
+    imgui_NewFrame();
+
+    bool result = ImGui::BeginMainMenuBar();
+    lua_pushboolean(L, result);
+
+    return 1;
+}
+
+static int imgui_EndMainMenuBar(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 0);
+    imgui_NewFrame();
+
+    ImGui::EndMainMenuBar();
+
+    return 0;
+}
+
+static int imgui_BeginMenu(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 1);
+    imgui_NewFrame();
+
+    const char* label = luaL_checkstring(L, 1);
+    bool enabled = lua_toboolean(L, 2);
+
+    if (!lua_isboolean(L, 2)) {
+        enabled = true;
+    }
```

Comment:
Why not:

bool enabled = true;
if (lua_isboolean(L, 2) {
    enabled = lua_toboolean(L, 2);
}

Or actually why even do the lua_isboolean() check at all? Could we not settle to only use lua_toboolean()?

Reply:
The `lua_isboolean()` check is there to make `enabled` default to `true` if not given, since `lua_toboolean()`on its own returns `false` even if the argument is `nil`. I've gone ahead and added your version of the check.

Reply:
Ah, gotcha. Makes sense!